dd-sdk-android/src/main/kotlin/com/datadog/android/log/internal/domain/LogGenerator.kt

```diff
@@ -48,21 +57,59 @@ internal class LogGenerator(
         bundleWithRum: Boolean = true,
         userInfo: UserInfo? = null,
         networkInfo: NetworkInfo? = null
-    ): Log {
+    ): LogEvent {
         val combinedAttributes = resolveAttributes(attributes, bundleWithTraces, bundleWithRum)
+        val formattedDate = synchronized(simpleDateFormat) {
```

Comment:
I am not sure why we were synchronizing on the `simpleDataFormat` so I decided to keep it as before. @xgouchet do you remember why were we doing this ?

Reply:
Yes, it's because the date format class is not thread safe and should be only used by one thread at a time.

Reply:
Also I am thinking that maybe we should use a static `formatter` here as this `LogGenerator` can be instantiated multiple times.

Reply:
In that case making it static will not be good for the perfs as a LogGenerator should be able to work independently from the others and having this static and synchronized will block all the LogGenerator instances. Thanks @xgouchet